app/src/main/java/com/poklad/androidifystore/utils/Resource.kt

```diff
@@ -1,7 +1,7 @@
 package com.poklad.androidifystore.utils
 
-sealed class Resource() {
-    class Success<T>(data: T) : Resource()
-    class Error(val throwable: Throwable?) : Resource()
-    object Loading : Resource()
+sealed class Resource<T>(val data: T? = null, val message: String? = null) {
+    class Success<T>(data: T?) : Resource<T>(data)
+    class Error<T>(message: String, data: T? = null): Resource<T>(data, message)
```

Comment:
why you need data in error? why do you need message in error? just pass throwable in error